app/routes.py

```diff
@@ -1 +1,164 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task 
+from app.models.goal import Goal 
+from flask import Blueprint, make_response, abort, request, jsonify
+from datetime import datetime
+from dotenv import load_dotenv
+load_dotenv()
+from app.helpers import *
+
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+
+#************Task Routes*****************
+
+
+@task_bp.route("", methods=["POST"])
+def create_task():
+    return create_new_model(Task, request.get_json())
+
+
+@task_bp.route("", methods=["GET"])
+def get_all():
+    task_query = Task.query
+    title_query = request.args.get("sort")
+
+    if title_query == "asc":
+        task_query = Task.query.order_by(Task.title.asc())
+    if title_query == "desc":
+        task_query = Task.query.order_by(Task.title.desc())
+
+    tasks = task_query.all()
+    task_response = [task.to_dict() for task in tasks]
+
+    return jsonify(task_response)
+
+
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    return read_one_model(Task, task_id)
+
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
We should have some error handling here in case the request_body does not have keys "title" or "description". If the client sent over a request body that had a key "task_title" then line 51 would throw an unhandled exception. We should assume any time we get data from a client that there might be invalid data that we need to handle.